### ISSUE TITLE
Change flow of using Azure Virtual Network Subnet in the Azure Kubernetes Service

### ISSUE THREAD

**Issue by mdanylyuk** [Azure/azure-container-networking#2434]
**Component: Azure CNI**

**Describe in detail the feature/behavior/change you'd like to see:**  

Hello guys,

Can we change the IP request batch size from 16 to 4? Also, change the minimum free IP count.

I have clusters that use "Azure CNI networking for dynamic allocation of IPs", with 2 node pools - each of them is linked to a separate subnet range that is dedicated to pods only.
If I set the pod count to 32 I lose 6 IPs because system pods, _azure-cns_/_azure-npm_/_cloud-node-manager_/_csi-azuredisk-node_/_csi-azurefile-node_/_kube-proxy_, use node`s IP. And I can allocate only 26 IPs on each node.

For example, I have subnet /24 for pods on the second node pool. Azure allows to me use only 251 IPs from this subnet. And if the batch is 16 it means that I can use only 240 of them (15*16=240) and after this all pods will not be able to start, to get a new IP.
So by default, I lose 11 IPs in this case.

In summary, I can use, correctly, 198 IPs. Because 7 nodes block 32 IPs, but only 26 can be used, and 1 node can block only 16 IPs (7*26+16=198)

All lost IPs cannot be used even for cluster scaling/upgrading.

Almost similar calculations you can do for any combination of subnet size and maximum pods.

**Orchestrator: Kubernetes (AKS)**  

**Operating System: Linux**  

**Anything else you would like to add:**  
[Miscellaneous information that will assist in solving the issue.]

In the attachments, you can find logs of azure-cns pod from the node that cannot get the initial batch of IPs
[azure-cns-6h9s4.log](https://github.com/Azure/azure-container-networking/files/13673800/azure-cns-6h9s4.log)

**Comment by mdanylyuk** [comment]
Small PoC with calculations of different combinations

network /24 251 IPs
pods count 32 - max pods 198 max nodes 7 (also it's 14 service pods for defender + 1 service pod for metric service)
pods count 38 - max pods 208 max nodes 7 (also it's 14 service pods for defender + 1 service pod for metric service)
pods count 16 - max pods 140 max nodes 14 (also it's 28 service pods for defender + 2 service pod for metric service)
pods count 22 - max pods 176 max nodes 11 (also it's 22 service pods for defender + 2 service pod for metric service)

**Comment by rbtr** [comment]
@mdanylyuk The batch size is not end-user configurable. If you need it changed on your cluster, you may file a support request to AKS.

**Comment by mdanylyuk** [comment]
@rbtr I cannot say that it has to be end-user configurable thing. I'd rather want to say that the current configuration is not optimal.
But thank you for the advice to ask the AKS team also.
